File: 30daysOFcode_Python/day 20.py
#!/bin/python3

import sys

n = int(input().strip())
a = list(map(int, input().strip().split(' ')))

# Write Your Code Here
def swap(a, x, y):
    t = a[x]
    a[x] = a[y]
    a[y] = t

def bubbleSort(a):
    n = len(a)
    allSwapCount = 0
    for i in range(0, n):
        swapCount = 0

        for j in range(0, n -i- 1):

            if a[j] > a[j + 1]:
                swap(a, j, j + 1)
                swapCount += 1
        if swapCount == 0:
            break
        else:
            allSwapCount += swapCount
    return (a[0], a[n - 1], allSwapCount)


first, last, swaps = bubbleSort(a)
print("Array is sorted in", swaps, "swaps.")
print("First Element:", first)

print("Last Element:", last)

File: 30daysOFcode_Julia/day 20.jl
# Enter your code here 
n = parse(Int, readline())
arr = parse.(Int,split(readline()))
numswaps = 0
for i = 1:n
    for j = 1:n-1
        if arr[j]>arr[j+1]
            dummy = arr[j]
            arr[j] = arr[j+1]
            arr[j+1] = dummy
            numswaps += 1
        end
    end
end
print("Array is sorted in $(numswaps) swaps.",'\n')
print("First Element: $(arr[1])",'\n')
print("Last Element: $(arr[end])",'\n')
